src/components/Initialization/Initialization.js

```diff
@@ -24,25 +24,32 @@ class Initialization extends PureComponent<Props, State> {
   };
 
   async componentDidMount() {
-    const nodeVersion = await getNodeJsVersion();
-
-    if (!nodeVersion) {
-      dialog.showErrorBox(
-        'Node missing',
-        'It looks like Node.js isn\'t installed. Node is required to use Guppy.\nWhen you click "OK", you\'ll be directed to instructions to download and install Node.'
-      );
-      shell.openExternal(
-        'https://github.com/joshwcomeau/guppy/blob/master/README.md#installation'
-      );
-    }
+    try {
+      await initializePath();
+
+      const nodeVersion = await getNodeJsVersion();
+
+      if (!nodeVersion) {

```

Comment:
I think we can throw an error here and move the dialog to the catch block, so we're displaying the dialog if `initializePath` fails or no `nodeVersion` found.

```suggestion
      if (!nodeVersion) {
        throw new Error('node-not-found');
      }
```

Reply:
mmh, I'm not sure about this as you're writing in `catch` that this can be swallowed. But If `initializePath` fails it won't do the `nodeVersion` check. Maybe we could always resolve from `initializePath` so the check will be always performed.

Reply:
Yeah, I could check the error in the catch block. I like your suggestion, will do that.